lib/gui/components/svg-icon.js

```diff
@@ -65,12 +67,13 @@ class SVGIcon extends react.Component {
     const width = this.props.width || DEFAULT_SIZE;
     const height = this.props.height || DEFAULT_SIZE;
 
-    const parser = new window.DOMParser();
-    const doc = parser.parseFromString(contents, 'image/svg+xml');
+    const img = document.createElement('img');
+    const doc = domParser.parseFromString(contents, 'image/svg+xml');
+    const parserError = doc.querySelector('parsererror');
     const svg = doc.querySelector('svg');
+    const svgXml = svg && _.isNil(parserError) ? svg.outerHTML : '';
```

Comment:
The thing is that its hard for the client code to "catch" these kinds of problems (there is no good messaging protocol apart from events). Can we allow this directive to take an array of embedded SVG strings, or an array of paths? We will try the first one, and if that fails, the next one, etc and if nothing works we throw an error.